lib/fetch/util.js

```diff
@@ -914,12 +912,37 @@ function isomorphicDecode (input) {
   // 1. To isomorphic decode a byte sequence input, return a string whose code point
   //    length is equal to input’s length and whose code points have the same values
   //    as the values of input’s bytes, in the same order.
-
-  if (input.length < MAXIMUM_ARGUMENT_LENGTH) {
-    return String.fromCharCode(...input)
+  const length = input.length
+  if ((2 << 15) - 1 > length) {
+    return String.fromCharCode.apply(null, input)
   }
-
-  return input.reduce((previous, current) => previous + String.fromCharCode(current), '')
+  let result = ''; let i = 0
+  // isTypedArray
+  if ('buffer' in input) {
+    let addition = (2 << 15) - 1
+    while (i < length) {
+      if (i + addition > length) {
+        addition = length - i
+      }
+      result += String.fromCharCode.apply(null, input.subarray(i, i += addition))
+    }
+    return result
+  }
+  // FIXME: Is it really possible that an array of numbers is passed on?
+  while (i + 32 < length) {
+    result += String.fromCharCode(
+      input[i], input[i + 1], input[i + 2], input[i + 3], input[i + 4], input[i + 5], input[i + 6], input[i + 7],
+      input[i + 8], input[i + 9], input[i + 10], input[i + 11], input[i + 12], input[i + 13], input[i + 14], input[i + 15],
+      input[i + 16], input[i + 17], input[i + 18], input[i + 19], input[i + 20], input[i + 21], input[i + 22], input[i + 23],
+      input[i + 24], input[i + 25], input[i + 26], input[i + 27], input[i + 28], input[i + 29], input[i + 30], input[i + 31]
+    )
+    i += 32
```

Comment:
It does the work but it feels like a codesmell. It also seems harder to maintain without a previous context of why we are doing that.
Is there any way to simplify this?

Reply:
I see two options: delete this part or delete the non-TypedArray part. Currently only the TypedArray seems to be used.

Reply:
It appears that only TypedArray is currently being used.
So this part can be removed.
Let me know if you have any other opinions.

Reply:
One idea is to comment out the code and raise an error if it is not a TypedArray.

Reply:
It would be easier to understand in the future if we did so.
```js
/**
 * @see https://infra.spec.whatwg.org/#isomorphic-decode
 * @param {Uint8Array} input
 */
function isomorphicDecode (input) {
  // 1. To isomorphic decode a byte sequence input, return a string whose code point
  //    length is equal to input’s length and whose code points have the same values
  //    as the values of input’s bytes, in the same order.
  const length = input.length
  if (!('buffer' in input)) {
    throw new TypeError('Nothing but TypedArray are supported. Please uncomment to make it work.')
  }
  if ((2 << 15) - 1 > length) {
    return String.fromCharCode.apply(null, input)
  }
  let result = ''; let i = 0
  // if ('buffer' in input) {
  let addition = (2 << 15) - 1
  while (i < length) {
    if (i + addition > length) {
      addition = length - i
    }
    result += String.fromCharCode.apply(null, input.subarray(i, i += addition))
  }
  return result
  // }
  // while (i + 32 < length) {
  //   result += String.fromCharCode(
  //     input[i], input[i + 1], input[i + 2], input[i + 3], input[i + 4], input[i + 5], input[i + 6], input[i + 7],
  //     input[i + 8], input[i + 9], input[i + 10], input[i + 11], input[i + 12], input[i + 13], input[i + 14], input[i + 15],
  //     input[i + 16], input[i + 17], input[i + 18], input[i + 19], input[i + 20], input[i + 21], input[i + 22], input[i + 23],
  //     input[i + 24], input[i + 25], input[i + 26], input[i + 27], input[i + 28], input[i + 29], input[i + 30], input[i + 31]
  //   )
  //   i += 32
  // }
  // // Decode the remaining characters.
  // while (i < length) {
  //   result += String.fromCharCode(input[i++])
  // }
  // return result
}
```

Reply:
@metcoder95 Doing this is easily accomplished.
```js
if (!('buffer' in input)) throw new TypeError("Supported only for TypedArray.")
```

Reply:
What I am currently thinking:
```js
function isomorphicDecode (input) {
  // 1. To isomorphic decode a byte sequence input, return a string whose code point
  //    length is equal to input’s length and whose code points have the same values
  //    as the values of input’s bytes, in the same order.
  const length = input.length
  if ((2 << 15) - 1 > length) {
    return String.fromCharCode.apply(null, input)
  }
  let result = ''; let i = 0
  // isTypedArray
  if ('buffer' in input) {
    let addition = (2 << 15) - 1
    while (i < length) {
      if (i + addition > length) {
        addition = length - i
      }
      result += String.fromCharCode.apply(null, input.subarray(i, i += addition))
    }
    return result
  }
  // while (i + 32 < length) {
  //   result += String.fromCharCode(
  //     input[i], input[i + 1], input[i + 2], input[i + 3], input[i + 4], input[i + 5], input[i + 6], input[i + 7],
  //     input[i + 8], input[i + 9], input[i + 10], input[i + 11], input[i + 12], input[i + 13], input[i + 14], input[i + 15],
  //     input[i + 16], input[i + 17], input[i + 18], input[i + 19], input[i + 20], input[i + 21], input[i + 22], input[i + 23],
  //     input[i + 24], input[i + 25], input[i + 26], input[i + 27], input[i + 28], input[i + 29], input[i + 30], input[i + 31]
  //   )
  //   i += 32
  // }
  // // Decode the remaining characters.
  // while (i < length) {
  //   result += String.fromCharCode(input[i++])
  // }
  // return result
  throw new TypeError("Supported only for TypedArray. If you want it to work, uncomment it.")
}
```

Reply:
> that's not code we want even if something other than a typed array was passed

Okay, I'll delete this part.